frontend/src/app/transactions/page.tsx

```diff
@@ -1,18 +1,218 @@
+import dayjs from 'dayjs';
+
+// Define the structure for filters
+interface FilterState {
+  action: string;
+  company: string;
+  dateRange: {
+    startDate: Date | null;
+    endDate: Date | null;
+  };
+}
+
+const INITIAL_VISIBLE_COUNT = 20;
+const LOAD_MORE_INCREMENT = 20;
+
+const TransactionsPage: React.FC = () => {
+  const { transactions, loading, error } = useTransactions();
+  const [filters, setFilters] = useState<FilterState>({
+    action: 'All',
+    company: '',
+    dateRange: {
+      startDate: null,
+      endDate: null,
+    },
+  });
+  const [visibleCount, setVisibleCount] = useState<number>(INITIAL_VISIBLE_COUNT);
+
+  // Handle filter updates
+  const updateFilter = <K extends keyof FilterState>(
+    key: K,
+    value: FilterState[K]
+  ) => {
+    setFilters((prev) => ({ ...prev, [key]: value }));
+    setVisibleCount(INITIAL_VISIBLE_COUNT);
+  };
+
+  // Handle date range selection
+  const handleDateChange = (
+    newValue: { startDate: Date | null; endDate: Date | null } | null
+  ) => {
+    if (newValue) {
+      setFilters((prev) => ({ ...prev, dateRange: newValue }));
+    } else {
+      setFilters((prev) => ({
+        ...prev,
+        dateRange: { startDate: null, endDate: null },
+      }));
+    }
+    setVisibleCount(INITIAL_VISIBLE_COUNT);
+  };
+
+  // Filter transactions based on filters
+  const filteredTransactions = useMemo(() => {
+    return transactions.filter((txn) => {
+      const txnDate = dayjs(txn.time_completed);
+
+      // Action Filter
+      const matchesAction =
+        filters.action === 'All' ||
+        txn.action.toLowerCase() === filters.action.toLowerCase();
+
+      // Company Filter
+      const matchesCompany =
+        !filters.company.trim() ||
+        txn.merchant_name.toLowerCase().includes(filters.company.toLowerCase());
+
+      // Date Range Filter
+      const { startDate, endDate } = filters.dateRange;
+      const afterStartDate =
+        !startDate || txnDate.isAfter(dayjs(startDate).subtract(1, 'day'));
+      const beforeEndDate =
+        !endDate || txnDate.isBefore(dayjs(endDate).add(1, 'day'));
+
+      return matchesAction && matchesCompany && afterStartDate && beforeEndDate;
+    });
+  }, [transactions, filters]);
+
+  // Transactions to display based on visible count
+  const displayedTransactions = useMemo(() => {
+    return filteredTransactions.slice(0, visibleCount);
+  }, [filteredTransactions, visibleCount]);
+
+  const hasMore = visibleCount < filteredTransactions.length;
+
+  const handleLoadMore = () => {
+    setVisibleCount((prev) => prev + LOAD_MORE_INCREMENT);
+  };
 
-export default function Transactions() {
   return (
-    <div className="mx-auto max-w-7xl">
-      <h1 className="mb-5 mt-5 text-center text-3xl font-extrabold text-black dark:text-white">
-        All recent transactions
+    <div className="mx-auto max-w-7xl p-6">
+      <h1 className="mb-8 text-center text-3xl font-extrabold text-gray-900 dark:text-white">
```

Comment:
Maybe use the class "mb-7 mt-10 text-5xl font-extrabold text-black dark:text-white" to match the about page?